test/UpdaterAndFilterEngineCreation.cpp

```diff
@@ -0,0 +1,172 @@
+/*
+ * This file is part of Adblock Plus <https://adblockplus.org/>,
+ * Copyright (C) 2006-present eyeo GmbH
+ *
+ * Adblock Plus is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * Adblock Plus is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Adblock Plus.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+
+#include <thread>
+#include <vector>
+#include <chrono>
+#include <string>
+#include <memory>
+
+#include "BaseJsTest.h"
+
+using namespace AdblockPlus;
+
+namespace
+{
+  class UpdaterAndFilterEngineCreationTest : public BaseJsTest
+  {
+  protected:
+
+    static const size_t COUNT = 100;
+    const std::string PROP_NAME = "patternsbackupinterval";
+    std::vector<std::thread> threadsVector;
+    Updater* updaterAddrArray[COUNT];
+    FilterEngine* filterAddrArray[COUNT];
+    DelayedWebRequest::SharedTasks webRequestTasks;
+    DelayedTimer::SharedTasks timerTasks;
+
+    void SetUp()
+    {
+      LazyFileSystem* fileSystem;
+      ThrowingPlatformCreationParameters platformParams;
+      platformParams.logSystem.reset(new LazyLogSystem());
+      platformParams.timer = DelayedTimer::New(timerTasks);
+      platformParams.fileSystem.reset(fileSystem = new LazyFileSystem());
+      platformParams.webRequest = DelayedWebRequest::New(webRequestTasks);
+      platform.reset(new Platform(std::move(platformParams)));
+      threadsVector.reserve(COUNT);
+      std::uninitialized_fill(updaterAddrArray, updaterAddrArray + COUNT, nullptr);
+      std::uninitialized_fill(filterAddrArray, filterAddrArray + COUNT, nullptr);
+    }
+
+    void CallGetUpdaterSimultaneously()
+    {
+      CallGetSimultaneously(true, false);
+    }
+
+    void CallGetFilterEngineSimultaneously()
+    {
+      CallGetSimultaneously(false, true);
+    }
+
+    void CallGetUpdaterAndGetFilterEngineSimultaneously()
+    {
+      CallGetSimultaneously(true, true);
+    }
+
+  private:
+
+    void CallGetSimultaneously(bool isUpdater, bool isFilterEngine)
+    {
+      for (size_t idx = 0; idx < COUNT; ++idx)
+        threadsVector.push_back(
+          std::thread([this, idx, isUpdater, isFilterEngine]() -> void {
+            Updater* updaterAddr = nullptr;
```

Comment:
Do we need these variables? It seems one can just assign an item in the array.

Reply:
I used them just to make the assignment in one place at the end of this method. But I can remove them.